### DISCUSSION TITLE
Servers initial connection to broker failing

### DISCUSSION THREAD

**Discussion by dom0317** [choria-io/general#46] [General]
I have a test setup with 1 client and 2 servers. Choria was manually deployed on all 3 nodes. That is the choria rpm, the nats, systemu and mco-rpc-support gems and the choria puppet modules were installed from command line. 

When I do choria discover or mco rpc rpcutil ping on the client it finds the client node but neither of the servers. 

On both servers in the choria.log I can see continual messages : 
{"component":"server","connection":"node2","identity":"node2","level":"warning","msg":"Initial connection to the Broker failed on try 47: dial tcp 192.168.0.42:4222: i/o timeout","time":"2024-01-26T12:14:23Z"}
That is the correct ip address for the client. And on the client doing a tcpdump on port 4222 I can see incoming messages from node2. I don't see any responses.

I don't see anything in the client choria.log (I have set logging to debug) to say why those connection attempts are rejected or timing out. 
Any idea what the problem could be, or what further investigation I can do?

**Comment by ripienaar** [comment]
io timeout tends to indicate your firewalls blocking the traffic or similar.  Can you test using telnet or nc to see the port is reachable?

**Reply by dom0317** [reply]
Yes, you're right. I had stopped iptables on the nodes but not on the client machine. D'oh! Once I did the nodes connected and choria discover works as expected. Thanks.
